funnel/models/membership.py

```diff
@@ -0,0 +1,436 @@
+
+    user_id = immutable(
+        db.Column(
+            None,
+            db.ForeignKey('user.id', ondelete='CASCADE'),
+            nullable=False,
+            index=True,
+        )
+    )
+    user = immutable(
+        db.relationship(
+            User,
+            foreign_keys=[user_id],
+            backref=db.backref(
+                'profile_crew_memberships',
+                lazy='dynamic',
+                cascade='all, delete-orphan',
+                passive_deletes=True,
+            ),
+        )
+    )
+
+    # Project crew roles (at least one must be True):
+
+    #: Editors can edit all common and editorial details of an event
+    is_editor = db.Column(db.Boolean, nullable=False, default=False)
+    #: Concierges are responsible for logistics and have write access
+    #: to common details plus read access to everything else. Unlike
+    #: editors, they cannot edit the schedule
+    is_concierge = db.Column(db.Boolean, nullable=False, default=False)
+    #: Ushers help participants find their way around an event and have
+    #: the ability to scan badges at the door
+    is_usher = db.Column(db.Boolean, nullable=False, default=False)
+
+    @declared_attr
+    def __table_args__(cls):
+        args = list(super(cls, cls).__table_args__)
+        args.append(
+            db.CheckConstraint(
+                'is_editor IS TRUE OR is_concierge IS TRUE OR is_usher IS TRUE',
+                name='project_crew_membership_has_role',
+            )
+        )
+        return tuple(args)
+
+    @property
+    def user_details(self):
+        return {'fullname': self.user.fullname, 'username': self.user.username}
+
+    @property
+    def edit_url(self):
+        return self.url_for('edit', _external=True)
+
+    @property
+    def delete_url(self):
+        return self.url_for('delete', _external=True)
+
+    def offered_roles(self):
+        """Roles offered by this membership record"""
+        roles = set()
+        if self.is_editor:
+            roles.add('project_editor')
+        if self.is_concierge:
+            roles.add('project_concierge')
+        if self.is_usher:
+            roles.add('project_usher')
+        return roles
+
+    def offered_roles_verbose(self):
+        roles = self.offered_roles()
+        return {role.replace('project_', '').title() for role in roles}
+
+    def roles_for(self, actor, anchors=()):
+        """Roles available to the specified actor and anchors"""
+        roles = super(ProjectCrewMembership, self).roles_for(actor, anchors)
+        roles.update(self.parent.roles_for(actor, anchors))
+        return roles
+
+
+# Project relationships: all crew, vs specific roles
+
+Project.active_crew_memberships = db.relationship(
+    ProjectCrewMembership,
+    lazy='dynamic',
+    primaryjoin=db.and_(
+        ProjectCrewMembership.project_id == Project.id,
+        ProjectCrewMembership.active,
+        ProjectCrewMembership.is_invite.is_(False),
```

Comment:
Use `~ProjectCrewMembership.is_invite`.

Using a comparison to False will result in this awkward condition: `(membership.is_invite == TYPE.INVITE) is False` whereas using `~` will result in `not (membership.is_invite == TYPE.INVITE)`, which PostgreSQL will (hopefully) automatically rewrite as `membership.is_invite != TYPE.INVITE)`

Reply:
Or maybe include an `is_not_invite` hybrid property.